src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -344,26 +345,35 @@ private bool TryBindUnconvertedBinaryOperatorToDefaultInterpolatedStringHandler(
 
             Debug.Assert(partsArrayBuilder.Count >= 2);
 
-            if (partsArrayBuilder.Count <= 4 && partsArrayBuilder.All(static parts => AllInterpolatedStringPartsAreStrings(parts)))
+            int count = 0;
+            bool allPartsAreStrings = true;
+
+            foreach (var parts in partsArrayBuilder)
             {
-                // This is case 2. Let the standard machinery handle it
-                partsArrayBuilder.Free();
-                return false;
+                count += parts.Length;
+                allPartsAreStrings &= AllInterpolatedStringPartsAreStrings(parts);
```

Comment:
>allPartsAreStrings [](http://example.com/codeflow?start=16&length=18)

It looks like accumulating this value is not necessary, we simply can inline ```AllInterpolatedStringPartsAreStrings``` call into the ``if``` condition. It also looks like asserting the local at the end doesn't add a lot of value.